plugins/inputs/snmp/snmp.go

```diff
@@ -535,8 +538,8 @@ func (t Table) Build(gs snmpConnection, walk bool) (*RTable, error) {
 		Time: time.Now(), //TODO record time at start
 		Rows: make([]RTableRow, 0, len(rows)),
 	}
-	for _, r := range rows {
-		if len(r.Tags) < tagCount {
+	for i, r := range rows {
+		if len(r.Tags)+emptyTags[i] < tagCount {
 			// don't add rows which are missing tags, as without tags you can't filter
 			continue
 		}
```

Comment:
No, the main difference from my perspective is that the defined tag is either empty or does not exist:

https://github.com/hahnjo/telegraf/commit/9699fcac21706619ddd33124626f01f17fd43ce0#diff-a240fd60347435c7253f4238fae424efL61 defines the test data. `.1.0.0.0.1.1.2` is just empty for `rtr3` while `.1.0.0.0.1.1.3` doesn't exist for `rtr4`. I'd consider the latter an error which should be filtered out...

Anyway, should I open a new pull request with the new branch if that's the way you prefer?